apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/PoolRewrite/SessionPoolOptions.cs

```diff
@@ -0,0 +1,233 @@
+﻿// Copyright 2017 Google Inc. All Rights Reserved.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Google.Api.Gax;
+using Google.Api.Gax.Grpc;
+using System;
+
+namespace Google.Cloud.Spanner.V1.PoolRewrite
+{
+    /// <summary>
+    /// Options for session pools.
+    /// </summary>
+    public sealed class SessionPoolOptions
+    {
+        // Note: if any of these defaults are changed, update the documentation accordingly.
+        private int _maximumActiveSessions = 100;
+        private int _minimumPooledSessions = 10;
+        private TimeSpan _idleSessionRefreshDelay = TimeSpan.FromMinutes(15);
+        private TimeSpan _poolEvictionDelay = TimeSpan.FromDays(7);
+        private ResourcesExhaustedBehavior _waitOnResourcesExhausted = ResourcesExhaustedBehavior.Block;
+        private double _writeSessionsFraction;
+        private int _timeout = 60;
+        private TimeSpan _maintenanceLoopDelay = TimeSpan.FromSeconds(30);
+        private int _maximumConcurrentSessionCreates = 10;
+        private RetrySettings.IJitter _sessionRefreshJitter = new ProportionalRandomJitter(0.1);
+        private RetrySettings.IJitter _sessionEvictionJitter = new ProportionalRandomJitter(0.1);
+        private IClock _clock = SystemClock.Instance;
+        private IScheduler _scheduler= SystemScheduler.Instance;
+
+        /// <summary>
+        /// Constructs a new <see cref="SessionPoolOptions"/> with default values.
+        /// </summary>
+        public SessionPoolOptions()
+        {
+        }
+
+        /// <summary>
+        /// Maximum number of sessions that can be active per database. An active session is one that has been
```

Comment:
So, potentially there could be pooled idle sessions and callers waiting but we couldn't paired them because we have already reached MaximumActiveSessions? That is, pooled idle sessions are not counted as ActiveSessions?